redis/src/cluster_slotmap.rs

```diff
@@ -1,28 +1,24 @@
+use std::sync::Arc;
+use std::sync::RwLock;
 use std::{
     collections::{BTreeMap, HashSet},
     fmt::Display,
     sync::atomic::AtomicUsize,
 };
 
-use crate::cluster_routing::{Route, Slot, SlotAddr, SlotAddrs};
+use dashmap::DashMap;
+
+use crate::cluster_routing::{Route, ShardAddrs, Slot, SlotAddr};
+
+pub(crate) type NodesMap = DashMap<Arc<String>, Arc<RwLock<ShardAddrs>>>;
```

Comment:
Question:

I don't have the entire context (GitHub folds existing code etc). But from what I have seen, we don't use `.write()` I have only seen `.read()` calls. Do we need this lock?

Reply:
You're correct, it hasn't been used in this PR - the write lock will be used in the next PR where i'll add support in topology updates based on MOVED errors. Then we will need to do modifications on the ShardAddrs object - for example when a MOVED error suggests a failover, or in scenarios when a replica is removed from existing ShardAddr as it is promoted to a primary in a different shard.